build.zig

```diff
@@ -18,17 +20,48 @@ fn concat(allocator: *std.mem.Allocator, str: []const u8, str2: []const u8) !std
 
 pub fn build(b: *Builder) void {
     src_files = ArrayList([]const u8).init(b.allocator);
+    src_files_asm = ArrayList([]const u8).init(b.allocator);
     const debug = b.option(bool, "debug", "build with debug symbols / make qemu wait for a debug connection") orelse false;
     var build_path = b.option([]const u8, "build-path", "path to build to") orelse "bin";
     var src_path = b.option([]const u8, "source-path", "path to source") orelse "src";
     var target = b.option([]const u8, "target", "target to build/run for") orelse "x86";
     const builtin_target = if (mem.eql(u8, target, "x86")) builtin.Arch.i386 else unreachable;
 
+    b.makePath(build_path) catch unreachable;
+    var grub_path = concat(b.allocator, build_path, "/iso/boot/grub") catch unreachable;
+    var kern_path = concat(b.allocator, build_path, "/kernel") catch unreachable;
+    var _a_path = concat(b.allocator, build_path, "/kernel/arch/") catch unreachable;
+    var a_path = concat(b.allocator, _a_path.toSlice(), target) catch unreachable;
+    b.makePath(grub_path.toSlice()) catch unreachable;
+    b.makePath(kern_path.toSlice()) catch unreachable;
+    b.makePath(a_path.toSlice()) catch unreachable;
+
     const iso_path = concat(b.allocator, build_path, "/pluto.iso") catch unreachable;
 
     src_files.append("kernel/kmain") catch unreachable;
 
     // Add the architecture init file to the source files
+    var arch_init = concat(b.allocator, "kernel/arch/", target) catch unreachable;
+    b.makePath("bin/kernel/arch") catch unreachable;
+
+    // Add the assemblies
+    const _arch_init = concat(b.allocator, arch_init.toSlice(), "/") catch unreachable;
+    const _arch_path = concat(b.allocator, src_path, "/") catch unreachable;
+    const arch_path = concat(b.allocator, _arch_path.toSlice(), arch_init.toSlice()) catch unreachable;
+
+    var arch_dir = os.Dir.open(b.allocator, arch_path.toSlice()) catch unreachable;
```

Comment:
It may potentially be better to add a check for the architecture and add files to `src_files_asm` based on that, something like:

```zig
switch (builtin_target) {
    builtin.Arch.i386 => {
        src_files_asm.append("src/kernel/arch/x86/irq_asm.s") catch unreachable;
        src_files_asm.append("src/kernel/arch/x86/isr_asm.s") catch unreachable;
    },
    else => {}
}
```
Since this is simpler and caters to the case when you may not want to assemble all .s files in the arch directory. What do you think?

Reply:
Well, being a computer scientist, i had to over engineer the hell out of it. But ur idea is much simpler